static/src/javascripts/bootstraps/enhanced/preferences.js

```diff
@@ -0,0 +1,123 @@
+// @flow
+import React from 'react/addons';
+import {
+    getPopularFiltered,
+    deleteFromSummary,
+    showInMegaNav,
+    showInMegaNavEnable,
+    showInMegaNavEnabled,
+} from 'common/modules/onward/history';
+import isEmpty from 'lodash/objects/isEmpty';
+
+const init = (): void => {
+    const placeholder: ?HTMLElement = document.getElementById(
+        'preferences-history-tags'
+    );
+
+    const initialiseSummaryTagsSettings = () => {
+        const SummaryTagsList: React = React.createClass({
+            getInitialState() {
+                return {
+                    popular: getPopularFiltered(),
+                };
+            },
+            handleRemove(tag: string) {
+                deleteFromSummary(tag);
+                this.setState({
+                    popular: getPopularFiltered({ flush: true }),
+                });
+                showInMegaNav();
+            },
+            render() {
+                const self = this;
+
+                const tags: Object = this.state.popular.reduce((obj, tag) => {
+                    obj[tag[0]] = React.DOM.span(
+                        {
+                            className:
+                                'button button--small button--tag button--secondary',
+                        },
+                        React.DOM.button(
+                            {
+                                onClick: self.handleRemove.bind(self, tag[0]),
+                                'data-link-name': `remove | ${tag[1]}`,
+                            },
+                            'X'
+                        ),
+                        React.DOM.a(
+                            {
+                                href: `/${tag[0]}`,
+                            },
+                            tag[1]
+                        )
+                    );
+                    return obj;
+                }, {});
+
+                let helperText;
+
+                if (isEmpty(tags)) {
```

Comment:
What do you think about `!tags || !Object.keys(tags).length`, so we could ditch the lodash dependency?